### PR TITLE
Documentation cleanup

### PR DESCRIPTION
**Pull request by outcatcher**
## Summary of the Pull Request
Cleanup and fix errors HCL code in documentation, examples and tests

## PR Checklist
* [x] Documentation updated.

### CODE REVIEW

**Review by lego963 — DISMISSED** [review]

**Review by lego963 — APPROVED** [review]

File: docs/data-sources/vpc_v1.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ The arguments of this data source act as filters for querying the available VPCs
 
 * `status` - (Optional) The current status of the desired VPC. Can be either CREATING, OK, DOWN, PENDING_UPDATE, PENDING_DELETE, or ERROR.
 
-* `name` - (Optional) A unique name for the VPC. The name must be unique for a tenant. The value is a string of no more than 64 characters and can contain digits, letters, 
+* `name` - (Optional) A unique name for the VPC. The name must be unique for a tenant. The value is a string of no more than 64 characters and can contain digits, letters,
   underscores (_), and hyphens (-).
```
Comment (anton-sidelnikov) [inline docs/data-sources/vpc_v1.md]: strange line break

File: docs/resources/identity_role_assignment_v3.md [hunk 2]
```diff
@@ -7,10 +7,10 @@ subcategory: "Identity and Access Management (IAM)"
 Manages a V3 Role assignment within group on OpentelekomCloud IAM Service.
 
 -> **Note:** You _must_ have admin privileges in your OpentelekomCloud cloud to use
-this resource. 
+this resource.
```
Comment (anton-sidelnikov) [inline docs/resources/identity_role_assignment_v3.md]: line break

File: docs/resources/rds_instance_v3.md [hunk 3]
```diff
@@ -201,17 +206,17 @@ The following arguments are supported:
 
 * `ha_replication_mode` - (Optional) Specifies the replication mode for the standby DB instance. For MySQL, the value
   is async or semisync. For PostgreSQL, the value is async or sync. For
-  Microsoft SQL Server, the value is sync. 
+  Microsoft SQL Server, the value is sync.
```
Comment (anton-sidelnikov) [inline docs/resources/rds_instance_v3.md]: and here

File: examples/basic-examples/modules/as/main.tf [hunk 4]
```diff
@@ -69,22 +69,22 @@ resource "opentelekomcloud_as_configuration_v1" "as_config_1" {
 ### create vpc
 
 resource "opentelekomcloud_vpc_v1" "vpc_as" {
-  name = "${var.vpc_name_as}-${random_id.as.id}"
-  cidr = "${var.vpc_cidr_as}"
+  name = var.vpc_name_as}-${random_id.as.id
```
Comment (anton-sidelnikov) [inline examples/basic-examples/modules/as/main.tf]: mistake here

File: examples/basic-examples/modules/as/main.tf [hunk 5]
```diff
@@ -69,22 +69,22 @@ resource "opentelekomcloud_as_configuration_v1" "as_config_1" {
...
+  gateway_ip    = var.subnet_gateway_ip1_as
+  vpc_id        = opentelekomcloud_vpc_v1.vpc_as.id
+  primary_dns   = var.primary_dns_as
+  secondary_dns = var.secondary_dns_as
 }
 
 ### create secuirty group
 resource "opentelekomcloud_networking_secgroup_v2" "secgroup_as" {
-  name        = "${var.secgroup_name_as}-${random_id.as.id}"
+  name        = var.secgroup_name_as}-${random_id.as.id
```
Comment (anton-sidelnikov) [inline examples/basic-examples/modules/as/main.tf]: mistake

File: examples/basic-examples/modules/ecs/main.tf [hunk 6]
```diff
@@ -2,46 +2,46 @@
...
-  region            = "${var.region}"
-  availability_zone = "${var.availability_zone}"
-  name              = "${var.ecs_name}-${random_id.ecs.id}"
-  image_id          = "${var.image_id}"
-  flavor_id         = "${var.flavor_id}"
-  key_pair          = "${var.key_name}"
-  security_groups   = ["${var.security_groups}"]
+  region            = var.region
+  availability_zone = var.availability_zone
+  name              = var.ecs_name}-${random_id.ecs.id
```
Comment (anton-sidelnikov) [inline examples/basic-examples/modules/ecs/main.tf]: mistake

File: examples/basic-examples/modules/ecs/main.tf [hunk 7]
```diff
@@ -60,22 +60,22 @@ resource "opentelekomcloud_networking_floatingip_v2" "floatip_1" {
...
-  region            = "${var.region}"
-  availability_zone = "${var.availability_zone}"
-  name              = "${var.ecs_name}-${random_id.ecs.id}"
-  #image_id         = "${var.image_id}"
-  flavor_id       = "${var.flavor_id}"
-  key_pair        = "${var.key_name}"
-  security_groups = ["${var.security_groups}"]
+  region            = var.region
+  availability_zone = var.availability_zone
+  name              = var.ecs_name}-${random_id.ecs.id
```
Comment (anton-sidelnikov) [inline examples/basic-examples/modules/ecs/main.tf]: here

File: examples/basic-examples/modules/ecs/main.tf [hunk 8]
```diff
@@ -98,17 +98,17 @@ resource "opentelekomcloud_blockstorage_volume_v2" "myvol" {
...
-  availability_zone = "${var.availability_zone}"
-  name              = "${var.ecs_name}-${random_id.ecs.id}"
-  #image_id         = "${var.image_id}"
-  
-  flavor_id       = "${var.flavor_id}"
-  key_pair        = "${var.key_name}"
-  security_groups = ["${var.security_groups}"]
+  region            = var.region
+  availability_zone = var.availability_zone
+  name              = var.ecs_name}-${random_id.ecs.id
```
Comment (anton-sidelnikov) [inline examples/basic-examples/modules/ecs/main.tf]: here

File: docs/data-sources/deh_server_v1.md [hunk 9]
```diff
@@ -13,7 +13,7 @@ variable "deh_id" { }
 variable "server_id" { }
 
 data "opentelekomcloud_deh_server_v1" "deh_server" {
-  id        = "${var.deh_id}",
+  id        = var.deh_id,
   server_id = "${var.server id}"
```
Comment (lego963) [inline docs/data-sources/deh_server_v1.md]: forgot here

File: examples/basic-examples/modules/iam/iam.tf [hunk 10]
```diff
@@ -6,7 +6,7 @@ resource "random_id" "project" {
 
 resource "opentelekomcloud_identity_project_v3" "project_1" {
   provider = "opentelekomcloud.iam"
-  name     = "${var.project_name}_${random_id.project.id}"
+  name     = var.project_name}_${random_id.project.id
```
Comment (anton-sidelnikov) [inline examples/basic-examples/modules/iam/iam.tf]: here

File: docs/resources/compute_volume_attach_v2.md [hunk 11]
```diff
@@ -44,12 +44,12 @@ resource "opentelekomcloud_compute_instance_v2" "instance_1" {
 
 resource "opentelekomcloud_compute_volume_attach_v2" "attachments" {
   count       = 2
-  instance_id = "${opentelekomcloud_compute_instance_v2.instance_1.id}"
+  instance_id = opentelekomcloud_compute_instance_v2.instance_1.id
   volume_id   = "${element(opentelekomcloud_blockstorage_volume_v2.volumes.*.id, count.index)}"
```
Comment (lego963) [inline docs/resources/compute_volume_attach_v2.md]: miss

File: docs/resources/css_cluster_v1.md [hunk 12]
```diff
@@ -20,7 +20,7 @@ resource "opentelekomcloud_css_cluster_v1" "cluster" {
   node_config {
     flavor = "css.medium.8"
     network_info {
-      security_group_id = "${opentelekomcloud_networking_secgroup_v2.secgroup.id}"
+      security_group_id = opentelekomcloud_networking_secgroup_v2.secgroup.id
       network_id        = "{{ network_id }}"
```
Comment (lego963) [inline docs/resources/css_cluster_v1.md]: add vars here pls

File: docs/resources/rds_instance_v3.md [hunk 13]
```diff
@@ -27,11 +27,11 @@ resource "opentelekomcloud_rds_instance_v3" "instance" {
     port     = "8635"
   }
 
-  security_group_id = "${opentelekomcloud_networking_secgroup_v2.secgroup.id}"
+  security_group_id = opentelekomcloud_networking_secgroup_v2.secgroup.id
   subnet_id         = "{{ subnet_id }}"
```
Comment (lego963) [inline docs/resources/rds_instance_v3.md]: add vars here

File: docs/resources/rds_instance_v3.md [hunk 14]
```diff
@@ -60,19 +60,19 @@ resource "opentelekomcloud_networking_secgroup_v2" "secgroup" {
 resource "opentelekomcloud_rds_instance_v3" "instance" {
   name              = "terraform_test_rds_instance"
   availability_zone = ["{{ availability_zone_1 }}", "{{ availability_zone_2 }}"]
```
Comment (lego963) [inline docs/resources/rds_instance_v3.md]: and here pls

File: docs/resources/rds_instance_v3.md [hunk 15]
```diff
@@ -60,19 +60,19 @@ resource "opentelekomcloud_networking_secgroup_v2" "secgroup" {
...
     password = "P@ssw0rd1!9851"
     type     = "PostgreSQL"
     version  = "9.5"
     port     = "8635"
   }
-  security_group_id   = "${opentelekomcloud_networking_secgroup_v2.secgroup.id}"
+  security_group_id   = opentelekomcloud_networking_secgroup_v2.secgroup.id
   subnet_id           = "{{ subnet_id }}"
-  vpc_id              = "{{ vpc_id }}" 
+  vpc_id              = "{{ vpc_id }}"
```
Comment (lego963) [inline docs/resources/rds_instance_v3.md]: and here

File: docs/resources/rds_instance_v3.md [hunk 16]
```diff
@@ -94,36 +94,41 @@ resource "opentelekomcloud_rds_instance_v3" "instance" {
...
+  name        = "terraform_test_security_group"
   description = "terraform security group acceptance test"
 }
 
 resource "opentelekomcloud_compute_floatingip_v2" "ip" {}
 
 resource "opentelekomcloud_rds_instance_v3" "instance" {
-  availability_zone = ["{{ availability_zone_1 }}", "{{ availability_zone_2 }}"]
+  availability_zone   = [
+    "{{ availability_zone_1 }}",
```
Comment (lego963) [inline docs/resources/rds_instance_v3.md]: plus one

File: docs/resources/rds_instance_v3.md [hunk 17]
```diff
@@ -94,36 +94,41 @@ resource "opentelekomcloud_rds_instance_v3" "instance" {
...
+    version  = "9.5"
+    port     = "8635"
   }
-  name = "terraform_test_rds_instance"
-  security_group_id = "${opentelekomcloud_networking_secgroup_v2.secgroup.id}"
-  subnet_id = "{{ subnet_id }}"
-  vpc_id = "{{ vpc_id }}" 
+  name                = "terraform_test_rds_instance"
+  security_group_id   = opentelekomcloud_networking_secgroup_v2.secgroup.id
+  subnet_id           = "{{ subnet_id }}"
```
Comment (lego963) [inline docs/resources/rds_instance_v3.md]: plus one

File: docs/resources/vpc_peering_accepter_v2.md [hunk 18]
```diff
@@ -17,47 +17,47 @@ connection into management.
...
 resource "opentelekomcloud_vpc_peering_connection_v2" "peering" {
   provider       = "opentelekomcloud.main"
-  name           = "${var.peer_name}"
-  vpc_id         = "${opentelekomcloud_vpc_v1.vpc_main.id}"
-  peer_vpc_id    = "${opentelekomcloud_vpc_v1.vpc_peer.id}"
-  peer_tenant_id =  "${var.tenant_id}"
+  name           = var.peer_name
+  vpc_id         = opentelekomcloud_vpc_v1.vpc_main.id
+  peer_vpc_id    = opentelekomcloud_vpc_v1.vpc_peer.id
+  peer_tenant_id =  var.tenant_id
```
Comment (lego963) [inline docs/resources/vpc_peering_accepter_v2.md]: remove 1 space pls after `=`

File: examples/app-with-networking/main.tf [hunk 19]
```diff
@@ -1,6 +1,6 @@
 resource "opentelekomcloud_compute_keypair_v2" "terraform" {
   name       = "terraform"
-  public_key = "${file("${var.ssh_key_file}.pub")}"
+  public_key = file("${var.ssh_key_file}.pub")
```
Comment (lego963) [inline examples/app-with-networking/main.tf:3]: `"${var.ssh_key_file}.pub"` miss here

File: examples/basic-examples/modules/as/main.tf [hunk 20]
```diff
@@ -5,24 +5,24 @@ resource "random_id" "as" {
 }
 ### create keypair
 resource "opentelekomcloud_compute_keypair_v2" "as-keypair" {
-  name       = "${var.key_name_as}-${random_id.as.id}"
-  public_key = "${file("public-2048.txt")}"
+  name       = var.key_name_as}-${random_id.as.id
```
Comment (lego963) [inline examples/basic-examples/modules/as/main.tf]: `}-${`

File: examples/basic-examples/modules/cce/cce.tf [hunk 21]
```diff
@@ -1,44 +1,44 @@
...
-      volumetype = "SATA"
-    }
-  ]
-
-  max_pods = 2
-
-}
+# test required params
+resource "opentelekomcloud_cce_cluster_v3" "cluster_1" {
+  name                   = "${var.cce_name}-1"
```
Comment (lego963) [inline examples/basic-examples/modules/cce/cce.tf:3]: `${` miss

File: examples/basic-examples/modules/ces/ces.tf [hunk 22]
```diff
@@ -5,16 +5,16 @@
...
+  alarm_description = var.alarm_desc
   metric {
     #"namespace" = "SYS.ECS"
     namespace   = "SYS.AS"
     metric_name = "as"
     dimensions {
       name = "AutoScalingGroup"
-      #"value" = "${opentelekomcloud_compute_instance_v2.webserver.id}"
-      value = "${var.as_group_id}"
+      #"value" = opentelekomcloud_compute_instance_v2.webserver.id
```
Comment (lego963) [inline examples/basic-examples/modules/ces/ces.tf:16]: should we remove this comment?

File: examples/basic-examples/modules/ces/ces.tf [hunk 23]
```diff
@@ -5,16 +5,16 @@
 #}
 
 resource "opentelekomcloud_ces_alarmrule" "alarm_rule" {
-  alarm_name        = "${var.alarm_name}"
-  alarm_description = "${var.alarm_desc}"
+  alarm_name        = var.alarm_name
+  alarm_description = var.alarm_desc
   metric {
     #"namespace" = "SYS.ECS"
```
Comment (lego963) [inline examples/basic-examples/modules/ces/ces.tf:11]: useless comment?

File: examples/basic-examples/modules/csbs/csbs.tf [hunk 24]
```diff
@@ -1,8 +1,8 @@
 
 resource "opentelekomcloud_csbs_backup_v1" "backup_v1" {
   backup_name   = "${var.backup_name}_csbs"
```
Comment (lego963) [inline examples/basic-examples/modules/csbs/csbs.tf:3]: can we remove interp here?

File: examples/basic-examples/modules/cts/cts.tf [hunk 25]
```diff
@@ -1,16 +1,16 @@
...
+  is_support_smn            = true
+  topic_id                  = "urn:smn:eu-de:b730519ca7064da2a3233e86bee139e4:smn-test"
+  is_send_all_key_operation = false
+  operations                = ["login", "create"]
+  need_notify_user_list     = ["user1"]
+
+}
+
+
+#data "opentelekomcloud_cts_tracker_v1" "tracker" {
```
Comment (lego963) [inline examples/basic-examples/modules/cts/cts.tf:14]: maybe remove this commented example?

File: examples/basic-examples/modules/dcs/dcs.tf [hunk 26]
```diff
@@ -11,12 +11,12 @@ data "opentelekomcloud_dcs_product_v1" "product_1" {
 
 resource "opentelekomcloud_dcs_instance_v1" "instance_1" {
   name              = "${var.dcs_name}_required"
```
Comment (lego963) [inline examples/basic-examples/modules/dcs/dcs.tf:13]: can we remove this interp? and comment on line 19

File: examples/basic-examples/modules/dcs/dcs.tf [hunk 27]
```diff
@@ -29,17 +29,17 @@ resource "opentelekomcloud_dcs_instance_v1" "instance_1" {
 }
 resource "opentelekomcloud_dcs_instance_v1" "instance_2" {
   name              = "${var.dcs_name}_redis"
```
Comment (lego963) [inline examples/basic-examples/modules/dcs/dcs.tf:31]: plus one
+42 line

File: examples/basic-examples/modules/ecs/main.tf (hunk 8 above)
Comment (lego963) [inline examples/basic-examples/modules/ecs/main.tf]: miss

File: examples/basic-examples/modules/evs/main.tf [hunk 28]
```diff
@@ -1,39 +1,39 @@
...
+  instance_id = var.ecs_id
+  volume_id   = opentelekomcloud_blockstorage_volume_v2.volume_1.id
 }
 
 #attach multiple volumes to ecs
 resource "opentelekomcloud_blockstorage_volume_v2" "volumes" {
   count             = 2
-  availability_zone = "${var.availability_zone}"
+  availability_zone = var.availability_zone
   name              = "${format("vol-%02d", count.index + 1)}"
```
Comment (lego963) [inline examples/basic-examples/modules/evs/main.tf]: miss `"{$`

File: examples/basic-examples/modules/sg/main.tf [hunk 29]
```diff
@@ -5,7 +5,7 @@ resource "random_id" "sg" {
 #create security group for ecs
 
 resource "opentelekomcloud_networking_secgroup_v2" "secgroup1" {
-  name        = "${var.secgroup_name}-${random_id.sg.id}"
+  name        = var.secgroup_name}-${random_id.sg.id
```
Comment (anton-sidelnikov) [inline examples/basic-examples/modules/sg/main.tf]: here

File: examples/basic-examples/modules/iam/iam.tf (hunk 10 above)
Comment (lego963) [inline examples/basic-examples/modules/iam/iam.tf]: miss

File: examples/basic-examples/modules/vpc_subnet/main.tf [hunk 30]
```diff
@@ -5,8 +5,8 @@ resource "random_id" "vpc" {
 
 #create VPC
 resource "opentelekomcloud_vpc_v1" "vpc_1" {
-  name = "${var.vpc_name}-${random_id.vpc.id}"
-  cidr = "${var.vpc_cidr}"
+  name = var.vpc_name}-${random_id.vpc.id
```
Comment (anton-sidelnikov) [inline examples/basic-examples/modules/vpc_subnet/main.tf]: here

File: examples/basic-examples/modules/sg/main.tf (hunk 29 above)
Comment (lego963) [inline examples/basic-examples/modules/sg/main.tf]: misstake

File: examples/basic-examples/modules/vpc_subnet/main.tf (hunk 30 above)
Comment (lego963) [inline examples/basic-examples/modules/vpc_subnet/main.tf]: misstake

File: examples/sample-cases/loadbalancer.tf [hunk 31]
```diff
@@ -1,39 +1,39 @@
...
+  count          = var.instance_count
   lb_method   = "ROUND_ROBIN"
-  listener_id = "${opentelekomcloud_lb_listener_v2.listener.id}"
+  listener_id = opentelekomcloud_lb_listener_v2.listener.id
 }
 
 resource "opentelekomcloud_lb_member_v2" "member" {
-  count         = "${var.instance_count}"
+  count         = var.instance_count
   address       = "${element(opentelekomcloud_compute_instance_v2.webserver.*.access_ip_v4, count.index)}"
```
Comment (anton-sidelnikov) [inline examples/sample-cases/loadbalancer.tf]: interpolation here

File: examples/sample-cases/keypairs.tf [hunk 32]
```diff
@@ -1,5 +1,5 @@
 resource "opentelekomcloud_compute_keypair_v2" "keypair" {
-  count      = "${var.instance_count}"
+  count      = var.instance_count
   name       = "${var.project}-terraform_key"
-  public_key = "${file("${var.ssh_pub_key}")}"
+  public_key = file("${var.ssh_pub_key)}"
```
Comment (lego963) [inline examples/sample-cases/keypairs.tf]: remove here

File: examples/sample-cases/loadbalancer.tf (hunk 31 above)
Comment (lego963) [inline examples/sample-cases/loadbalancer.tf]: remove here pls

File: examples/sample-cases/volumes.tf [hunk 33]
```diff
@@ -1,7 +1,7 @@
 resource "opentelekomcloud_blockstorage_volume_v2" "volume" {
   count = "${var.disk_size_gb > 0 ? var.instance_count : 0}"
```
Comment (lego963) [inline examples/sample-cases/volumes.tf]: can we remove here?

File: examples/vbs.tf [hunk 34]
```diff
@@ -13,29 +13,29 @@ resource "opentelekomcloud_vbs_backup_policy_v2" "vbs" {
 }
 
 data "opentelekomcloud_vbs_backup_policy_v2" "policies" {
-  id = "${opentelekomcloud_vbs_backup_policy_v2.vbs.id}"
+  id = opentelekomcloud_vbs_backup_policy_v2.vbs.id
 }
 
 resource "opentelekomcloud_blockstorage_volume_v2" "volume" {
   count = "${var.disk_size_gb > 0 ? var.instance_count : 0}"
```
Comment (lego963) [inline examples/vbs.tf]: remove here pls

**Review by lego963 — CHANGES_REQUESTED** [review]
format terraform files in examples pls

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]
comments

File: examples/app-with-networking/main.tf (hunk 19 above)
Comment (outcatcher) [reply examples/app-with-networking/main.tf:3]: No, this is valid place for using interpolation

File: examples/basic-examples/modules/ces/ces.tf (hunk 22 above)
Comment (outcatcher) [reply examples/basic-examples/modules/ces/ces.tf:16]: I was not removing comments, examples still to be checked

File: examples/basic-examples/modules/ces/ces.tf (hunk 23 above)
Comment (outcatcher) [reply examples/basic-examples/modules/ces/ces.tf:11]: I was not removing comments, examples still to be checked

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by lego963 — DISMISSED** [review]

**Review by lego963 — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**